lib/report.js

```diff
@@ -201,9 +208,14 @@ const tapTestCaption = (id, test) => {
 class TapReporter extends Reporter {
   constructor(options) {
     super(options);
+    // TODO(lundibundi): fix (remove) this once TSR properly supports require from worker_threads
     if (this.options.type && this.options.type !== 'tap') {
-      // TODO: do .pipe(this.options.stream) as soon as TSR supports it
-      this.options.stream = new TSR(this.options.type);
+      this.log = (...args) => {
+        // TODO: do .pipe(this.options.stream) as soon as TSR supports it
+        this.options.stream = new (lazyTSR())(this.options.type);
```

Comment:
Couldn't we just define a getter on the `stream` property instead of rewriting the `log()` method?

Reply:
I assume you want to use `Object.defineProperty` on `this.options`? I think this will only complicate the code without any merit.

Reply:
Not necessarily `Object.defineProperty()`, you can also define a getter in another way.
Redefining the `log()` method twice complicates the code even more in my opinion.